### ISSUE TITLE
Section 2 - Host Application Form

### ISSUE THREAD

**Issue by pardhama** [hackforla/HomeUniteUs#509]
### The Problem:
<Strike>Currently, Hosts have no means to provide information about themselves and their homes to the homeless helping organization online.</Strike>
Currently, the Host lack an automated way to share information about themselves and their homes to the organization or the coordinator.

### User Story:
<Strike>As a Host, I should be able to see the questions in the application form and fill it to the best of my knowledge for the matching process</strike>
As a Host, I should be able to see the questions in the Profile (formerly application form) and fill it to the best of my knowledge for the matching process.

###High-Level Approach:
<Strike>Application form page with multiple questions and a place to enter the answers to those questions. The page should also contain options like save and have the ability to move to any section of the application.</strike>
Profile (formerly application form) page with multiple questions and a repository to enter the answers. The page should include the ability to save the progress made and to navigate to any profile section.

Additional note: Host will see all communication in their process as "Profile", but we will internally refer to the form that the Host fill out as "Intake Profile" in future stories to distinguish this from the "Matching Profile" that gets generated once the "Intake Profile" is approved. In this user story, "Profile" and "Intake Profile" are synonymous.

### Considerations
Anticipate that users will be using desktop, tablet, mobile devices --> use responsive design
In the future, the plan is for the host onboarding experience for multiple organizations (SPY and beyond), where different organizations might want to modify the application form needed for a Hosts.

### Goals & Success
<Strike>Success is if a host can successfully complete the application form and edit it based on the comments received by the coordinator.</strike>
The host can successfully access, navigate and view the high-level task groups steps, action items and status on the  dashboard in one centralized area. 

### Metrics
Number of applications completed vs the number of applications started.
Average time taken to complete each application

### Key Features:
- Text-based questions for most questions (Questions where the answer would be in paragraph type). (As of now application form is same as safe place for youth. The link is in the PDF below)
- Date, phone number, email, checkbox and signature question types/validations
- Navigation on all profile (application form) pages to enable the users to modify their answers and move to any section of the profile (application)
- ~Save button on each page to save the profile (formerly application) and exit~
- Autosave feature every time Host tries to exit the page they are currently on with a message that informs them that the page's progress will be saved
- Ability to see the same saved profile (formerly application) after logging in 
- Ability to edit the answers multiple times before completing the profile.
- Required field validations and indications that required fields have not been filled out
- Ability to see a final overall preview before submitting the profile
- ~Ability to sign the profile before each submission~
- Button to submit the final profile
- Ability to get a PDF version of the signed profile (formerly application) on submission (download button or email whichever is easier; quick feedback from eng is that render + download pdf shouldn't be a major effort, will need input on what the pdf should look like). Update:
    - Email should be auto generated on submission
    - Ability to get PDF version at any given time from the Guest dashboard
- Ability to edit the answers if requested by the coordinator after a profile (application) is submitted.
- Ability to see the last version of the submitted profile (need to work out the design for accessing/displaying profiles)

### Acceptance Criteria:

- Once a Host is logged in, they should be able to start the profile from the dashboard page under the Profile section.
- On the first page of the profile (formerly application form), a Host should be able to see the minimum expectations (Just like we see terms and conditions)
- A Host should be able to type/modify the answers to questions until the profile is submitted.
- Host should see an error if a question type is filled incorrectly (Date, phone number, email and signature)
- Should be able to fill the profile (formerly application form) in multiple sessions by saving the profile using the autosave functionality (The autosaving functionality will add all the filled questions into the database for this particular guest)
- Given the Host is trying to close the tab, move to a different section of the profile or click any navigation item that would take them away from the current page they are on, they should see a pop up confirming if they want to leave and inform them that changes would be autosaved
- Host should be able to see pre-filled answers for any answers that were saved at any point.
- Host should see a (*) for mandatory questions and they should not be allowed to submit the profile if a mandatory question is not completed. They should see an error describing that a mandatory question is not completed and highlight the missing question (easier to identify) when they try to navigate away from a page
- Host should be able to see a preview of all answers before the final submission.
- If a Host is logged in and has exited before submitting the profile (formerly application form), they should be able to continue the profile from the dashboard page under the application section and they should be redirected towards the first page that has an unfilled required field.
- Host should have a submit profile (formerly application) button at the end of the application form which will lock the profile and send the profile for review to the coordinator.
- Host should have the ability to get a PDF version of the signed profile on submission (download button or email whichever is easier)
- Once the Host has completed the profile (formerly application form), they should see the profile status as Pending Review on the dashboard page.
- If a coordinator finds any question incomplete and requests more info (to be discussed):
- The Host should see a Needs more info status on the application tab of the dashboard and would receive an email/call from the coordinator on what needs to be edited
- Profile should be unlocked for the Host to edit all fields
- Host should be required to sign the edited version of the profile before they click submit.
- Host should be able to navigate to any section of the profile (formerly application) without completing previous sections

### Host application form:
[Host application form Link](https://drive.google.com/file/d/0B3o4rAE_DB1IeTFGX3J3ZjNPblo3OE13a0xrX28xVW5hOHRv/view?usp=share_link&resourcekey=0-WBt4S-NTHGrIR7tc9-qEDQ)

### Design

- [ ] Review existing designs on Figma
- [ ] Edit existing designs as needed
- [ ] Designs are edited and ready for engineering. Link to the Figma: (share Figma link here when ready)

### Team Reviews:

- [ ] Product - 
- [ ] Design - Help Needed
- [ ] Engineering -

**Comment by baipai** [comment]
I've made the host dashboard adjust to the guest dashboard.

**Comment by baipai** [comment]
I am trying to figure out how users can navigate from the dashboard to the application and how they can return to the dashboard if they choose to. What was the original plan? I asked Ed about this last week but still want to double-check. 

And when I review the user story, where or how does the coordinator leave comments about their application? I couldn't find it in the guest section. What's the plan for this?


cc: @sanya301 @edela0015 @rpradheap

**Comment by baipai** [comment]
Note: As the guest part is still being worked on, the host part will continue until things are somewhat finalized.

**Comment by lola3736** [comment]
@jasmineg95 @sm10949 lets discuss during our Thursday team meeting, is this ready for design discussion?

**Comment by londondkim** [comment]
https://github.com/hackforla/HomeUniteUs/issues/508#issuecomment-2451133915

**Comment by londondkim** [comment]
Duplicate confirmed by Lola's comment from 11/21/24

**Comment by lola3736** [comment]
@londondkim this is most updated host intake profile issue but it appears you had moved it from "In Progress" to "Legacy Tickets", can you please clarify, thanks much.
cc - @pixellady

**Comment by londondkim** [comment]
@lola3736  >> comment you made on Nov 21, 2024 notes that this issue was replaced with #508 
https://github.com/hackforla/HomeUniteUs/issues/509#issuecomment-2492987119

**Comment by londondkim** [comment]
@lola3736 On a second look, it looks like this profile mentions "dashboard". Maybe this was not a duplicate but intended to serve as another ticket to create dashboard? Prior to Profile Intake page. I would use this as a reference or may even replace my dashboard ticket with this one? Let me know what you think
